### ISSUE TITLE
[BUG] Global teardown is running multiple times on sharding setup

### ISSUE THREAD

**Issue by Nav-2d** [microsoft/playwright#23627]
### System info
- Playwright Version: 1.34.3
- Operating System: Mac
- Browser: All
- Other info:

### Source code

- [ ] I provided exact source code that allows reproducing the issue locally.

**Link to the GitHub repository with the repro**

[bug-repo](https://github.com/Nav-2d/pw-codegen-repro)

**Steps**
- Run the script `npm run all`. This runs the tests in parallel in two shards

**Expected**

The global teardown should run only once similar to global teardown

**Actual**

The global teardown runs in both the shards.

```
Running 4 tests using 1 worker, shard 2 of 2


Running 4 tests using 1 worker, shard 1 of 2

     1 [setup] › global.setup.ts:3:5 › run global setup
  ✓  1 [setup] › global.setup.ts:3:5 › run global setup (1.6s)
  ✓  1 [setup] › global.setup.ts:3:5 › run global setup (1.9s)
**global setup ran**
     2 [webkit] › example.spec.ts:4:7 › has title @smoke › has title
  ✓  2 [webkit] › example.spec.ts:4:7 › has title @smoke › has title (1.0s)
  ✓  2 [chromium] › example.spec.ts:4:7 › has title @smoke › has title (1.6s)
     3 [chromium] › example.spec.ts:9:7 › has title @smoke › has 12123
  ✓  3 [chromium] › example.spec.ts:9:7 › has title @smoke › has 12123 (809ms)
  ✓  4 [teardown] › global.teardown.ts:3:5 › run global teardown (1.6s)
**global teardown ran**

  ✓  4 [teardown] › global.teardown.ts:3:5 › run global teardown (1.4s)
**global teardown ran**

```

**Comment by aslushnikov** [comment]
@Nav-2d Both `global-steup` and `globa-teardown` are actually run twice; the reason you see only one `global-setup` log is because our line reporter is likely to overwrite the output.

I just ran it locally:

<img width="1264" alt="image" src="https://github.com/microsoft/playwright/assets/746130/382f978b-5ee7-47f6-b192-0446967b9778">

You can also check that global-setup is executed with the `line` reporter (it's more careful towards the terminal output). 

So all in all, works as intended.

**Comment by Nav-2d** [comment]
@aslushnikov - Thanks for looking at it. I guess my question is then - shouldn't global setup and teardown should run only once? (the word would mean run once during a test run). My use case:

Shard 1 - global setup, 3 tests are running, global teardown runs (this cleans up data for my application to reset it back to clean state)
Shard 2 - 4 tests are running but they are long and takes bit longer to run, global teardown runs 

Eventually when global teardown runs for first shard it actually ends up in cleaning the data that is required in shard 2.

**Comment by aslushnikov** [comment]
@Nav-2d shards are supposed to be launched on separate machines, so that's why the `global setup` and `global teardown` have to be run for each of them.

In your usecase, I'd recommend just runnings tests with more workers: there should be no reason to use shards when running on the same machine.

**Comment by aslushnikov** [comment]
> Would it make to sense to create a feature request to run setup and teardown only once during sharding setup? 

I don't think so: this fundamentally doesn't sound like a "right" solution to me.

As a work-around, you can limit the setup/teardown to only run once on a given machine by creating a marker files somewhere on the file system. Would this work?
